### PR TITLE
Replace jit test method  in favor of dynamo in `BaseTester`

### PR DESCRIPTION
**Pull request by johnnv1**
Updated  the `test/color/test_gray.py` as example

### CODE REVIEW

File: kornia/testing/__init__.py [hunk 1]
```diff
@@ -119,9 +119,9 @@ def test_exception(self, device, dtype):
     def test_cardinality(self, device, dtype):
         raise NotImplementedError("Implement a stupid routine.")
 
-    @abstractmethod
-    def test_jit(self, device, dtype):
-        raise NotImplementedError("Implement a stupid routine.")
+    # TODO: add @abstractmethod
+    def test_dynamo(self, device, dtype, torch_optimizer):
+        pass
```
Comment (johnnv1) [inline kornia/testing/__init__.py:124]: Maybe have a warning here to encourage the community to update the tests? It will get very verbose

**Comment by johnnv1** [comment]
macOS seems to not fully support dynamo, maybe it has other setup to do? 

https://github.com/kornia/kornia/actions/runs/3802730492/jobs/6468489701#step:4:13310

**Comment by johnnv1** [comment]
This is an active bug (this PR should fix it https://github.com/pytorch/pytorch/pull/89262) -- for now, should we disable the CI on macOS?

File: conftest.py [hunk 2]
```diff
@@ -63,6 +63,18 @@ def dtype(dtype_name) -> torch.dtype:
     return TEST_DTYPES[dtype_name]
 
 
+@pytest.fixture(scope='session')
+def torch_optimizer():
+    try:
+        import torch._dynamo as dynamo
+
+        torch.set_float32_matmul_precision('high')
+        return dynamo.optimize('inductor')
```
Comment (johnnv1) [inline conftest.py:72]: If we care with which backed is used here in tests, we can add the backend as an option (like dtype)

**Comment by johnnv1** [comment]
Maybe just run with torch nightly for dynamo tests `pytest ... -k 'test_dynamo'`

**Comment by johnnv1** [comment]
Hey @edgarriba @ducha-aiki, we can discuss/go ahead on this here to start dropping support for jit?

File: kornia/testing/__init__.py (hunk 1 above)
Comment (edgarriba) [reply kornia/testing/__init__.py:124]: I’d avoid that

File: conftest.py (hunk 2 above)
Comment (edgarriba) [reply conftest.py:72]: Yeah, we can improve later

**Comment by edgarriba** [comment]
> Hey @edgarriba @ducha-aiki, we can discuss/go ahead on this here to start dropping support for jit?

For me this approach goes well

File: conftest.py [hunk 3]
```diff
@@ -63,6 +63,18 @@ def dtype(dtype_name) -> torch.dtype:
...
+    except Exception as er:
+        del er
```
Comment (edgarriba) [inline conftest.py:74]: why delete  here?

**Review by edgarriba — APPROVED** [review]
LGTM. We make the process iterate and add slowly

File: conftest.py (hunk 3 above)
Comment (johnnv1) [reply conftest.py:74]: to not have a bare exception and not have an unused var too -- flake8 complains

**Comment by johnnv1** [comment]
I will check in a separated PR, the failing test (it's not related with the changes here)

```out
FAILED test/geometry/transform/test_image_registrator.py::TestImageRegistrator::test_registration_real[cpu-float32-loftr_homo] - AssertionError: Tensor-likes are not close!
```

**Pull request merged**